src/components/LeftSideBar.js

```diff
@@ -91,15 +92,24 @@ class LeftSideBar extends Component {
 
   // stores attribute selected
   handleClickAttribute = attribute => {
-    /*   if (this.selectedAttributes !== []) { //this needs to be replaced with add and remove
-            this.setState({
-                selectedAttributes: []
-            });
-            console.log('Clicked!!', attribute, this.state);
-        } else */ {
-      console.log('Clicked!!', attribute, this.state);
-      this.setState({ selectedAttributes: attribute.property_id }); // this.state.selectedAttributes.push(attribute.property_id) });
+    //console.log('====' + this.state.selectedAttributes);
+    //console.log(attribute.property_id);
+    let newArr = this.state.selectedAttributes;
+    if (_.includes(newArr, attribute.property_id)) {
+      //this needs to be replaced with add and remove
+
+      // uncomment this to remove. I commented this because added item is getting removed on second call
+      _.remove(newArr, elem => {
+        return elem === attribute.property_id;
+      });
+    } else {
+      newArr.push(attribute.property_id);
+      // this.state.selectedAttributes.push(attribute.property_id) });
     }
+    this.setState({
+      selectedAttributes: newArr
+    });
+    console.log(this.state);
```

Comment:
Because of the async nature of `this.setState`, `console.log` on this line might not reflect the latest state. If you want to make sure what's the state after the call on line 109. Try this:
```
this.setState({
  selectedAttributes: newArr
}, () => console.log(this.state));
```